integrations/crowdnode/src/main/java/org/dash/wallet/integrations/crowdnode/utils/CrowdNodeBalanceCondition.kt

```diff
@@ -25,15 +25,15 @@ import org.dash.wallet.common.services.LeftoverBalanceException
 class CrowdNodeBalanceCondition {
     fun check(
         walletBalance: Coin,
-        address: Address,
+        address: Address?,
         amount: Coin,
         crowdNodeConfig: CrowdNodeConfig
     ) {
         runBlocking { // TODO: remove runBlocking when this class is used from Kotlin code only
             val crowdNodeBalance = crowdNodeConfig.getPreference(CrowdNodeConfig.LAST_BALANCE) ?: 0
-            val crowdNodeAddress = CrowdNodeConstants.getCrowdNodeAddress(address.parameters)
+            val crowdNodeAddress = address?.let { CrowdNodeConstants.getCrowdNodeAddress(address.parameters) }
 
-            if (crowdNodeBalance <= 0 && address != crowdNodeAddress) {
+            if (crowdNodeBalance <= 0 && (address == null || address != crowdNodeAddress)) {
```

Comment:
Coinbase doesn't always know where it will send the DASH before checking all conditions.
I suppose if we don't know the destination address, then it's unlikely to be the CrowdNode address.